### PR TITLE
feat(Alura): new presence

### PR DESCRIPTION
**Pull request by MadsonMendes**
## Description 

Created a presence for Alura as a request in student's server. Added some dynamic elements as well. Planning to translate everything to Portuguese later. 
## Acknowledgements
- [X] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [X] I linted the code by running `yarn format`
- [X] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>
  

image  

image  

image  

image  

image  
</details>

### CODE REVIEW

**Comment by ImRodry** [comment]
The website is in portuguese so the presence should be too. I don’t think you should have multilang

File: websites/A/Alura/presence.ts [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+	) {
+		presenceData.details = "Viewing a course"; // alura.viewCourse
+		// Check in what course page you are, and return the same course name
+		if (
+			!document.querySelector(
+				"div.container.course-header-banner-content > div > div.hreview-aggregate > h1 > strong"
+			)
+		) {
+			title = document.querySelector("section.task-menu-header > div > a > h2");
+			presenceData.state = title.textContent;
```
Comment (EncryptedDev) [inline websites/A/Alura/presence.ts:48]:
```suggestion
			presenceData.state = document.querySelector("section.task-menu-header > div > a > h2").textContent;
```

File: websites/A/Alura/presence.ts [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+		} else {
+			title = document.querySelector(
+				"div.container.course-header-banner-content > div > div.hreview-aggregate > h1 > strong"
+			);
+			presenceData.state = title.textContent;
```
Comment (EncryptedDev) [inline websites/A/Alura/presence.ts:53]:
```suggestion
			presenceData.state = document.querySelector(
				"div.container.course-header-banner-content > div > div.hreview-aggregate > h1 > strong"
			).textContent;
```

File: websites/A/Alura/presence.ts [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+			title = document.querySelector(".intro__title");
+			presenceData.state = `"${title.textContent}"`;
+		}
+	} else if (document.location.pathname.includes("/scubadev")) {
+		//ScubaDev
+		presenceData.largeImageKey = "scubadev";
+		presenceData.details = 'Listening to "ScubaDev"'; // alura.scubadev
+		if (
+			document.querySelector(".podcast-header-title") !== null &&
+			document.querySelector(".--margin-left-image") === null
```
Comment (EncryptedDev) [inline websites/A/Alura/presence.ts:196]: Update this on future references of `!== null` and `=== null`
```suggestion
			document.querySelector(".podcast-header-title") &&
			!document.querySelector(".--margin-left-image")
```

**Review by EncryptedDev — CHANGES_REQUESTED** [review]
Hello! I've noticed the following imperfections with your PR, please address them :+1:

**Review by EncryptedDev — COMMENTED** [review]
Hello 👋

- Run `yarn format` as the metadata file is not formatted with tabs

File: websites/A/Alura/dist/metadata.json [hunk 4]
```diff
@@ -0,0 +1,29 @@
+{
+	"$schema": "https://schemas.premid.app/metadata/1.7",
+	"author": {
+		"name": "Mad",
+		"id": "653742649798295560"
+	},
+	"service": "Alura",
+	"description": {
+		"en": "The largest online technology school in Brazil.",
+		"pt_BR": "A maior escola online de tecnologia do Brasil."
```
Comment (fr0r) [inline websites/A/Alura/dist/metadata.json:10]: 👍

**Review by fr0r — APPROVED** [review]

File: websites/A/Alura/presence.ts [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+		else if (
+			document
+				.querySelector("a.course-header-banner-breadcrumb__category")
+				.getAttribute("aria-label") === "Mobile"
+		)
+			presenceData.largeImageKey = "https://i.imgur.com/mLpf7Yz.png";
+		else if (
+			document
+				.querySelector("a.course-header-banner-breadcrumb__category")
+				.getAttribute("aria-label") === "Inovação & Gestão"
```
Comment (ImRodry) [inline websites/A/Alura/presence.ts:88]: This should only be computed once so you should make it into a constant

File: websites/A/Alura/presence.ts [hunk 6]
```diff
@@ -0,0 +1,250 @@
...
+			"Data Science"
+		)
+			presenceData.largeImageKey = "https://i.imgur.com/4AQNRB1.png";
+		else if (
+			document.querySelector("div.formacao__info-categoria > a").textContent ===
+			"Mobile"
+		)
+			presenceData.largeImageKey = "https://i.imgur.com/mLpf7Yz.png";
+		else if (
+			document.querySelector("div.formacao__info-categoria > a").textContent ===
```
Comment (ImRodry) [inline websites/A/Alura/presence.ts:130]: Ditto

File: websites/A/Alura/presence.ts [hunk 7]
```diff
@@ -0,0 +1,250 @@
...
+			"Inovação & Gestão"
+		)
+			presenceData.largeImageKey = "https://i.imgur.com/RrC0z8A.png";
+	} else if (document.location.pathname.includes("/courses"))
+		presenceData.details = "Vendo meus cursos";
+	else if (document.location.pathname.includes("/user")) {
+		if (document.querySelector(".profile-header-name")) {
+			presenceData.details = strings.user;
```
Comment (ImRodry) [inline websites/A/Alura/presence.ts:138]: Why are strings being used here? That makes no sense

File: websites/A/Alura/presence.ts [hunk 8]
```diff
@@ -0,0 +1,250 @@
...
+			presenceData.state = document.querySelector(
+				".profile-header-name"
+			).textContent;
+		} else {
+			if (document.location.pathname.includes("/planos-de-estudo"))
+				presenceData.details = "Vendo planos de estudo";
+			else if (document.location.pathname.includes("/payments"))
+				presenceData.details = "Vendo pegamentos";
```
Comment (ImRodry) [inline websites/A/Alura/presence.ts:146]: Pegamentos é noutro lado…

File: websites/A/Alura/presence.ts [hunk 9]
```diff
@@ -0,0 +1,250 @@
...
+			else document.location.pathname.includes("/rank");
+			presenceData.details = "Vendo o ranking";
```
Comment (ImRodry) [inline websites/A/Alura/presence.ts:148]: De quê?

File: websites/A/Alura/presence.ts [hunk 10]
```diff
@@ -0,0 +1,250 @@
...
+		presenceData.details = "Vendo o Aluraflix";
+	} else if (document.location.pathname.includes("/carreirasemfronteiras")) {
+		presenceData.largeImageKey = "https://i.imgur.com/XbsVEFV.png";
+		presenceData.details = 'Ouvindo o "Carreiras sem Fronteiras"';
+		if (
+			document.querySelector(".podcast-header-title") &&
+			!document.querySelector(".--margin-left-image")
+		) {
+			presenceData.state = `"${
+				document.querySelector(".podcast-header-title").textContent
```
Comment (ImRodry) [inline websites/A/Alura/presence.ts:161]: Don’t compute this twice

File: websites/A/Alura/presence.ts [hunk 11]
```diff
@@ -0,0 +1,250 @@
...
+			}"`;
+		}
+	} else if (document.location.pathname.includes("/cases")) {
+		presenceData.largeImageKey = "https://i.imgur.com/rtevoAV.png";
+		presenceData.details = "Assistindo um Case";
```
Comment (ImRodry) [inline websites/A/Alura/presence.ts:166]: Case não deve estar em maiúscula

File: websites/A/Alura/presence.ts [hunk 12]
```diff
@@ -0,0 +1,250 @@
...
+		if (
+			document.querySelector(".intro__title") &&
+			document.location.pathname.includes("/extra")
+		) {
+			presenceData.state = `"${
+				document.querySelector(".intro__title").textContent
```
Comment (ImRodry) [inline websites/A/Alura/presence.ts:172]: Don’t compute this twice

File: websites/A/Alura/presence.ts [hunk 13]
```diff
@@ -0,0 +1,250 @@
...
+			document.querySelector(".podcast-header-title") &&
+			!document.querySelector(".--margin-left-image")
+		) {
+			presenceData.state = `"${
+				document.querySelector(".podcast-header-title").textContent
+			}"`;
+		}
+	} else if (document.location.pathname.includes("/hipsterstech")) {
+		presenceData.largeImageKey = "https://i.imgur.com/T7SMsQU.png";
+		presenceData.details = 'Ouvindo o "Hipsters.tech"';
```
Comment (ImRodry) [inline websites/A/Alura/presence.ts:210]: O outro tinha T maiúsculo e este já não?

**Review by ImRodry — COMMENTED** [review]
Here’s a proper review of the strings + code. I didn’t comment on all lines but there is a lot of unnecessary computation being done that should be avoided

**Review by EncryptedDev — APPROVED** [review]

**Comment by EncryptedDev** [comment]
Ignore the comments, my apologies

**Review by Bas950 — APPROVED** [review]

**Pull request merged**